python/dask_cudf/dask_cudf/sorting.py

```diff
@@ -186,8 +186,14 @@ def quantile_divisions(df, by, npartitions):
         dtype = df[columns[0]].dtype
         divisions = divisions[columns[0]].astype("int64")
         divisions.iloc[-1] += 1
-        divisions = sorted(
-            divisions.drop_duplicates().astype(dtype).values.tolist()
+        divisions = (
+            sorted(
+                divisions.dropna()
+                .drop_duplicates()
+                .astype(dtype)
+                .values.tolist()
+            )
+            + [None] * divisions.null_count
         )
```

Comment:
Can we use `.to_arrow` and sort, that way we are tampering the null ordering?

Reply:
Were you thinking something like

```python
        divisions = (
            sorted(
                divisions.drop_duplicates()
                .astype(dtype)
                .to_arrow().tolist()
            )
```

In that case, the null ordering is still being tampered by `drop_duplicates`, which places nulls first. From there we can't sort the resulting list as it contains `None`, unless you meant using an arrow sorting method before calling `tolist()`?

Reply:
I see, didn't notice the `drop_duplicates` call, yea `drop_duplicates` results in non-deterministic ordering. But `.to_arrow().tolist()` seems cleaner to me.

Reply:
Okay, in that case do you have any preference for how we handle sorting the list with nulls? Doing a quick search, this seems like a suitable solution:

```python
# https://stackoverflow.com/questions/18411560/sort-list-while-pushing-none-values-to-the-end

sorted(..., key=lambda x: (x is None, x))
```

Reply:
Try `sort_indices` of pyarrow, it seems to be doing what you want to do: https://arrow.apache.org/docs/python/generated/pyarrow.compute.sort_indices.html

If that's not possible the StackOverflow approach looks fine to me.